spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jProperties.java

```diff
@@ -166,6 +173,11 @@ private void configureDriverWithDefaults(DriverConfiguration driverConfiguration
 			driverConfiguration.setDriverClassName(EMBEDDED_DRIVER);
 			return;
 		}
+		if (ClassUtils.isPresent(BOLT_DRIVER, this.classLoader)) {
+			driverConfiguration.setDriverClassName(BOLT_DRIVER);
+			driverConfiguration.setURI( DEFAULT_BOLT_URI );
+			return;
+		}
```

Comment:
That's a breaking in behaviour. If you didn't specify anything and you upgrade, you're using bolt instead of http. I need to better understand the impact of this change before considering the code as is.